examples/rcbd_two_var.R

```diff
@@ -0,0 +1,179 @@
+yield <- c(6, 7, 9, 13, 16, 20, 8, 8, 9,
+           7, 8, 8, 12, 17, 18, 10, 9, 12,
+           9, 9, 9, 14, 18, 21, 11, 12, 11,
+           8, 10, 10, 15, 16, 22, 9, 9, 9)
+block <- gl(4, 9)
+clone <- rep(gl(3, 3, labels = c('c1', 'c2', 'c3')), 4)
+nitrogen <- rep(gl(3, 1, labels = c('n1', 'n2', 'n3')), 12)
+my.data <- data.frame(block, clone, nitrogen, yield)
+write.csv(my.data, 'rcbd_two_var.csv', quote=FALSE, row.names=FALSE)
+
+# Load the data.
+#-----------------------------------------------------------------------------#
+my.data <- read.csv('rcbd_two_var.csv')
+my.data$block <- as.factor(my.data$block)
+#-----------------------------------------------------------------------------#
+
+# Use a log10 transformation.
+#-----------------------------------------------------------------------------#
+my.data$log10.yield <- log10(my.data$yield)
+#-----------------------------------------------------------------------------#
+
+# Construct the model.
+#-----------------------------------------------------------------------------#
+model <- aov(log10.yield ~ block + clone + nitrogen + clone:nitrogen,
+             data=my.data)
+#-----------------------------------------------------------------------------#
+
+# Create box plot of levels of treatments for each factor (in this case clone &
+# nitrogen levels).
+dev.new()
+#-----------------------------------------------------------------------------#
+par(mfrow = c(2, 1))
+boxplot(log10.yield ~ clone, data = my.data,
+        main = "Effect of clone on yield",
+        xlab = "Clone", ylab = "log10(Yield)")
+boxplot(log10.yield ~ nitrogen, data = my.data,
+        main = "Effect of nitrogen on yield",
+        xlab = "Nitrogen Level", ylab = "log10(Yield)")
+#-----------------------------------------------------------------------------#
+invisible(dev.copy(png, 'rcbd-two-var-box-plots.png'))
+invisible(dev.off())
+
+# Plot two standard fit plots: residuals vs predicted, Normal Q-Q plot of the
+# residuals.
+dev.new()
+#------------------------------------------------------------------------------#
+par(mfrow = c(2, 1), oma = c(0, 0, 2, 0))
+plot(model, c(1, 2))
+#------------------------------------------------------------------------------#
+invisible(dev.copy(png, 'rcbd-two-var-fit-plots.png'))
+invisible(dev.off())
+
+# Make sure the residuals are normal (this can also be seen in the Q-Q plot).
+cat('Shapiro-Wilk Normality Test\n')
+sep(79)
+#------------------------------------------------------------------------------#
+shapiro.test(residuals(model))
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Run Levene's Test for a one-way ANOVA of each of the main factors.
+# TODO : clone is significant in Levene's test so transformation is necessary.
+cat("Levene's Test\n")
+sep(79)
+#------------------------------------------------------------------------------#
+leveneTest(yield ~ clone, data=my.data)
+leveneTest(yield ~ nitrogen, data=my.data)
```

Comment:
Noticed these are still using the original units, and not log10. It seems like it's corrected in the app at least for this example.